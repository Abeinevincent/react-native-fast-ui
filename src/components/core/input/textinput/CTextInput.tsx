import React from 'react';
import { Platform, type TextStyle, TextInput } from 'react-native';
import type { ICTextInput } from './CTextInput.type';
import { textStyleMapper } from '../../../utils/stylemapper';

const CTextInput: React.FC<ICTextInput> = ({
  style: additionalStyles,
  fs,
  c,
  w,
  br,
  ph,
  bw,
  bc,
  placeholder,
  ...rest
}) => {
  // Create style object for default styles
  const textInputStyle: TextStyle = {
    fontSize: fs ? fs : 16,
    color: c ? c : '#222',
    borderColor: bc ? bc : '#999',
    borderWidth: bw ? bw : 0.5,
    width: w ? w : '90%',
    borderRadius: br ? br : 50,
    paddingHorizontal: ph ? ph : 20,
  };

  // Iterate through props and apply corresponding style props dynamically
  Object.keys(rest).forEach((prop) => {
    const key = prop as keyof ICTextInput;
    if (textStyleMapper()[key]) {
      // @ts-ignore
      textInputStyle[textStyleMapper()[key]] = rest[key];
    }
  });

  // Apply platform-specific styles
  if (Platform.OS === 'android') {
    textInputStyle.includeFontPadding = rest.ifp ?? true;
    textInputStyle.verticalAlign = rest.va ?? 'auto';
  } else if (Platform.OS === 'ios') {
    textInputStyle.writingDirection = rest.wd ?? 'auto';
  }

  // Combine custom styles with additional styles
  const combinedStyles = [textInputStyle, additionalStyles];

  // Render CText component with combined styles and children
  return (
    <TextInput
      placeholder={placeholder ? placeholder : 'Default CTextInput component..'}
      placeholderTextColor={rest.placeholderTextColor || '#999'}
      style={combinedStyles}
      {...rest}
      testID="text-input"
    />
  );
};

export { CTextInput };
