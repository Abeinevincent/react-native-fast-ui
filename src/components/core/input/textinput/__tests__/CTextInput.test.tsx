import React from 'react';
import '@testing-library/jest-native/extend-expect'; // Import this to extend Jest's expect functionality
import { render } from '@testing-library/react-native';
import { CTextInput } from '../CTextInput';

describe('CTextInput Component', () => {
  test('renders correctly with default props', () => {
    const { getByTestId, toJSON } = render(<CTextInput testID="text-input" />);
    const textInput = getByTestId('text-input');
    // Add assertions for default styles and props
    expect(textInput).toHaveStyle({
      fontSize: 16,
      color: '#222',
      width: '90%',
      borderRadius: 50,
      paddingHorizontal: 20,
      borderWidth: 0.5,
      borderColor: '#999',
    });
    expect(toJSON).toMatchSnapshot();
  });

  test('renders with custom styles and props', () => {
    const { getByTestId } = render(
      <CTextInput
        testID="text-input"
        fs={18}
        c="blue"
        w={200}
        h={40}
        br={10}
        ph={15}
        bw={1}
        bc="#555"
        placeholder="Enter text here"
      />
    );
    const textInput = getByTestId('text-input');

    // Add assertions for custom styles and props
    expect(textInput).toHaveStyle({
      fontSize: 18,
      color: 'blue',
      width: 200,
      borderRadius: 10,
      paddingHorizontal: 15,
      borderWidth: 1,
      borderColor: '#555',
    });
  });

  test('handles placeholder correctly', () => {
    const { getByTestId } = render(
      <CTextInput testID="text-input" placeholder="Test Placeholder" />
    );
    const textInput = getByTestId('text-input');

    // Add assertions for placeholder text
    expect(textInput.props.placeholder).toBe('Test Placeholder');
  });

  // Add more tests as needed for specific functionality or user interactions
});
