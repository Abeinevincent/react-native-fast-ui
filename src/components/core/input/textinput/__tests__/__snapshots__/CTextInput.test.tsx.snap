// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTextInput Component renders correctly with default props 1`] = `[Function]`;
